packages/logging-bunyan/src/index.js

```diff
@@ -133,6 +134,30 @@ LoggingBunyan.prototype.write = function(record) {
     );
   }
 
+  // Stackdriver Log Viewer picks up the summary line from the 'message' field
+  // of the payload. Unless the user has provided a 'message' property also,
+  // move the 'msg' to 'message'.
+  if (!record.message) {
+    // Clone the object before modifying it.
+    record = extend({}, record);
+
+    // If this is an error, report the full stack trace. This allows Stackdriver
+    // Error Reporting to pick up errors automatically (for severity 'error' or
+    // higher). In this case we leave the 'msg' property intact.
+    // https://cloud.google.com/error-reporting/docs/formatting-error-messages
+    //
+    // TODO(ofrobots): when resource.type is 'global' we need to additionally
+    // provide serviceContext.service as part of the entry for Error Reporting
+    // to automatically pick up the error.
+    if (record.err && record.err.stack) {
+      record.message = record.err.stack;
+    } else if (record.msg) {
```

Comment:
Perhaps it would have been better for me to make separate PRs for winston and bunyan.

In case you're uncomfortable about winston having different behaviour, let's look at what winston's own built-in transports do for ambiguous cases:

```
winston.info('foo', new Error('bar'));
//info: foo Error: bar
//   at Object.<anonymous> (/Users/ofrobots/tmp/log-test/index.js:16:21)
// ->  simple concatenation.

winston.info(new Error('goo'))
//info:  Error: goo
//    at Object.<anonymous> (/Users/ofrobots/tmp/log-test/index.js:15:14)
// -> a simple concatenation: an empty `msg` is concatenated with the Error
```

I can make our concatenation even simpler (i.e. get rid of the ': ') to match this behaviour.

I still have strong feelings that Bunyan should have different behaviour.

Do you still have reservations?